train.py

```diff
@@ -544,9 +545,11 @@ def run(cfg: Any) -> None:
     model.to(cfg.environment._device)
 
     if cfg.architecture.force_embedding_gradients:
-        for param in model.backbone.base_model.get_input_embeddings().parameters():
-            param = param.float()
-            param.requires_grad = True
+        for module in model.modules():
+            if isinstance(module, torch.nn.Embedding):
+                for param in module.parameters():
+                    param.requires_grad = True
+                    param.data = param.data.float()
```

Comment:
What's the reason to cast to float explicitly?

Reply:
Mixed precision can't operate on float16 weights, so it will fail. I could only implement it for LORA+MP, but I think for stability it does not hurt to always cast this to float in this case.